cpp/src/rolling/rolling.cu

```diff
@@ -131,8 +134,12 @@ void gpu_rolling(gdf_size_type nrows,
     gdf_index_type const out_mask_location = cudf::util::detail::bit_container_index<bit_mask_t, gdf_index_type>(i);
 
     // only one thread writes the mask
-    if (0 == threadIdx.x % warpSize)
+    if (0 == threadIdx.x % warpSize) {
       out_col_valid[out_mask_location] = result_mask;
+      // Perform null bitmask null count
+      valid_count_accumulate = __popc(result_mask);
+      atomicAdd(output_null_count, valid_count_accumulate);
```

Comment:
This is one way to do it, but not the fast way -- this is one atomic per warp. The faster way is probably to keep the per-warp valid sum in shared memory, then do a cub::warp_reduce to get a per-block sum and do one atomic per block. See: https://github.com/rapidsai/cudf/blob/c68835079a70d0ea912854f5b7beff4792e4f6a1/cpp/src/stream_compaction/copy_if.cuh#L199-L212

We should probably generalize this across algorithms.

Reply:
Thanks @harrism for reviewing this PR. I think @jrhemstad has similar suggestion. So, I have used `single_lane_popc_block_reduce` which essentially stores  the per-warp valid sum in shared memory and then does a reduce. Please take another look.